src/three-components/ARRenderer.ts

```diff
@@ -52,93 +47,66 @@ export class ARRenderer extends EventDispatcher {
   public reticle: Reticle = new Reticle(this.camera);
   public raycaster: Raycaster|null = null;
 
-  private[$outputCanvas]: HTMLCanvasElement|null = null;
   private[$outputContext]: WebGLRenderingContext|null = null;
   private[$rafId]: number|null = null;
   private[$currentSession]: XRSession|null = null;
   private[$refSpace]: XRCoordinateSystem|null = null;
   private[$presentedScene]: ModelScene|null = null;
   private[$resolveCleanup]: ((...args: any[]) => void)|null = null;
-  private[$device]: XRDevice|null = null;
-  private[$devicePromise]: Promise<void|XRDevice>;
 
   /**
    * Given an inline Renderer, construct an ARRenderer and return it
    */
   static fromInlineRenderer(renderer: Renderer) {
-    return new ARRenderer(renderer.canvas, renderer.context!);
+    return new ARRenderer(renderer);
   }
 
-  constructor(
-      inputCanvas: HTMLCanvasElement, inputContext: WebGLRenderingContext) {
+  constructor(parentRenderer: Renderer) {
     super();
+    const inputContext: WebGLRenderingContext = parentRenderer.context!;
+    this.parentRenderer = parentRenderer;
 
-    this.renderer = null;
+    this.renderer = parentRenderer.renderer;
 
-    this.inputCanvas = inputCanvas;
     this.inputContext = inputContext;
 
     this.camera.matrixAutoUpdate = false;
 
     this.scene.add(this.reticle);
     this.scene.add(this.dolly);
-
-    // NOTE: XRDevice is being removed
-    // @see https://github.com/immersive-web/webxr/pull/405
-    this[$devicePromise] = this.resolveDevice()
-                               .then((device) => {
-                                 return this[$device] = device;
-                               })
-                               .catch((error) => {
-                                 console.warn(error);
-                                 console.warn('Browser AR will be disabled');
-                               });
   }
 
   initializeRenderer() {
-    if (this.renderer != null) {
-      return;
-    }
-
-    this.renderer = new WebGLRenderer(
-        {canvas: this.inputCanvas, context: this.inputContext});
-    this.renderer.setSize(window.innerWidth, window.innerHeight);
     this.renderer.setPixelRatio(1);
```

Comment:
If we're going to reuse our renderer, we'll need to reset this to the device DPR when we're done.

Reply:
I added a comment in postSessionCleanup. Renderer.render auto-adjusts the device pixel ratio as needed already.